editor/project_settings_editor.cpp

```diff
@@ -298,10 +318,13 @@ void ProjectSettingsEditor::_device_input_add() {
 		}
 	}
 
+	Dictionary event;
+	event["player"] = InputMap::PLAYER_1;
+	event["event"] = ie;
 	if (idx < 0 || idx >= events.size()) {
-		events.push_back(ie);
+		events.push_back(event);
 	} else {
-		events[idx] = ie;
+		events[idx] = event;
```

Comment:
This is causing the player index to be reset when you change an input key.

I would suggest this logic instead:
```
	if (idx < 0 || idx >= events.size()) {
		Dictionary event;
		event["player"] = player;
		event["event"] = ie;
		events.push_back(event);
	} else {
		Dictionary event = events[idx];
		event["event"] = ie;
	}
```